kratos/spaces/ublas_space.h

```diff
@@ -306,35 +307,34 @@ class UblasSpace
      * @param rA The matrix to compute the Jacobi norm
      * @return aux_sum: The Jacobi norm
      */
-    static TDataType JacobiNorm(MatrixType const& rA)
+    static TDataType JacobiNorm(const Matrix& rA)
     {
         TDataType aux_sum = TDataType();
-
-#ifndef _OPENMP
-        for (int i = 0; i < static_cast<int>(rA.size1()); i++)
-        {
-            for (int j = 0; j < static_cast<int>(rA.size2()); j++)
-            {
-                if (i != j)
-                {
+        #pragma omp parallel for reduction(+:aux_sum)
+        for (int i=0; i<static_cast<int>(rA.size1()); ++i) {
+            for (int j=0; j<static_cast<int>(rA.size2()); ++j) {
+                if (i != j) {
                     aux_sum += std::abs(rA(i,j));
                 }
             }
         }
-#else
-        #pragma omp parallel for reduction(+:aux_sum)
-        for (int i = 0; i < static_cast<int>(rA.size1()); i++)
-        {
-            for (int j = 0; j < static_cast<int>(rA.size2()); j++)
-            {
-                if (i != j)
-                {
-                    aux_sum += std::abs(rA(i,j));
+        return aux_sum;
+    }
+
+    static TDataType JacobiNorm(const compressed_matrix<TDataType>& rA)
+    {
+        TDataType aux_sum = TDataType();
+
+        typedef typename compressed_matrix<TDataType>::const_iterator1 t_it_1;
+        typedef typename compressed_matrix<TDataType>::const_iterator2 t_it_2;
+
+        for (t_it_1 it_1 = rA.begin1(); it_1 != rA.end1(); ++it_1) {
+            for (t_it_2 it_2 = it_1.begin(); it_2 != it_1.end(); ++it_2) {
+                if (it_2.index1() != it_2.index2()) {
+                    aux_sum += std::abs(*it_2);
```

Comment:
no clue how/why this works, but it seems to do the job
From: http://www.guwi17.de/ublas/doc/html/ex__sparse1_8cpp-source.html
@RiccardoRossi maybe you can take a look